### ISSUE TITLE
Save results into nexus format.

### ISSUE THREAD

**Issue by YooSunYoung** [scipp/ess#236]
The result of first data reduction step should be saved into nexus format to be fed to the next process.

It can be implemented once scipp/essnmx#15 is finished.

# Required Fields
- [x] Sample name
- [x] Distance from source to sample
- [x] Other source information
- [x] Detector origen(?)
- [x] Fast axes (in beamline frame(?))
- [x] Slow axes (in beamline frame(?))
- [x] Proton charge (?)
- [x] Detector counts (grouped weights/counts, gzip compressed)
- [x] Time bins (in [ms])
- [x] Pixel IDs
- [x] Crystal Rotation

# Questions
- Is there anything can be already done in the first reduction step instead...?
- Many fields used to be hard-coded and not straightforward where to retrieve them from the McStas file.
  Should users be able to update them manually some how...?
